test/Microsoft.AspNetCore.Routing.Tests/DefaultLinkGeneratorTest.cs

```diff
@@ -785,50 +748,6 @@ public void GetLink_OptionalParameter_ParameterNotPresentInValues()
             Assert.Equal("/Home/Index", link);
         }
 
-        [Fact]
-        public void GetLink_OptionalParameter_ParameterPresentInValuesAndDefaults()
-        {
-            // Arrange
-            var endpoint = CreateEndpoint(
-                template: "{controller}/{action}/{name?}",
-                defaultValues: new { name = "default-products" });
-            var linkGenerator = CreateLinkGenerator();
-            var context = CreateRouteValuesContext(
-                suppliedValues: new { action = "Index", controller = "Home", name = "products" });
-
-            // Act
-            var link = linkGenerator.GetLink(
-                httpContext: null,
-                new[] { endpoint },
-                context.ExplicitValues,
-                context.AmbientValues);
-
-            // Assert
-            Assert.Equal("/Home/Index/products", link);
-        }
-
-        [Fact]
-        public void GetLink_OptionalParameter_ParameterNotPresentInValues_PresentInDefaults()
```

Comment:
These two tests are not valid, and the fact that there were possible to write is a bug that I've fixed in RoutePattern. You can't have a default value for an optional parameter.